### DISCUSSION TITLE
Maximum time to keep alive idle browser connection

### DISCUSSION THREAD

**Discussion by popadko** [reactphp/reactphp#570] [Q&A]
Hi!

I am curious about the [$maximumTimeToKeepAliveIdleConnection](https://github.com/reactphp/http/blob/8111281ee57f22b7194f5dba225e609ba7ce4d20/src/Io/ClientConnectionManager.php#L39) value which is used inside the Browser. Do I understand correctly that it is a 1 millisecond? 

Typically such value should be seconds. In my case, the requests are coming less often than 1 millisecond, so the connections are not reused.

But maybe I am missing something. Please help to figure it out.

**Comment by SimonFrings** [comment]
Hey @popadko, thanks for bringing this up :+1:

> Do I understand correctly that it is a 1 millisecond?

Yes you understand correctly, we intentionally set this to the default of 1 millisecond back when we introduced this feature in reactphp/http#486. @clue used the following reasoning in his PR:

> […] Additionally, idle connections will currently only be kept alive for a maximum of 1ms between requests to make sure consumers of this library will not be affected negatively. By using a very short timeout period, the loop does not keep running noticeably longer than without this feature. This doesn't matter for long running scripts, but it's important for short running scripts to not appear to be "blocked". In the future, we should give more control over this timeout to allow reusing connections also for common use cases of periodically polling hosts using a longer interval. […]

As you can see, there are plans to give more control over this and make this value adjustable for each use case. We're currently working towards the next major version for [ReactPHP (v3)](https://github.com/orgs/reactphp/discussions/481), can't promise that we'll add this for the release, but will definitely see this as a candidate. Even if this shouldn't be part of the v3 release, I think it will find its way into the project shortly after.

If you need this feature on shorter notice, you can also reach out to us or support the ReactPHP ecosystem by [becoming a sponsor](https://github.com/sponsors/reactphp) ❤️
